DataFormats/PatCandidates/src/TriggerObjectStandAlone.cc

```diff
@@ -306,3 +327,126 @@ void TriggerObjectStandAlone::unpackPathNames(const edm::TriggerNames &names) {
     pathNames_.swap(paths);
 }
 
+void TriggerObjectStandAlone::packFilterLabels(const std::vector<std::string> &names)  {
+    if (!filterLabelIndices_.empty()) {
+        throw cms::Exception("RuntimeError", "Error, trying to pack filter labels for an already packed TriggerObjectStandAlone");
+    }
+    std::vector<std::string> unmatched;
+    std::vector<uint16_t> indices;
+    indices.reserve(filterLabels_.size());
+   
+    auto start = names.begin(), fail = names.end();
+    for (unsigned int i = 0, n = filterLabels_.size(); i < n; ++i) {
+        auto match = std::lower_bound(names.begin(), names.end(), filterLabels_[i]);
+        if (match != fail && *match == filterLabels_[i]) {
+            indices.push_back(match - start);
+        } else {
+            static std::atomic<int> _warn(0);
+            if(++_warn < 5) edm::LogWarning("TriggerObjectStandAlone::packFilterLabels()") << "Warning: can't resolve '" << filterLabels_[i] << "' to a label index. idx: " << i <<std::endl;
+            unmatched.push_back(std::move(filterLabels_[i]));
+        }
+    }
+    std::sort(indices.begin(), indices.end()); // try reduce enthropy
+    filterLabelIndices_.swap(indices);
+    filterLabels_.swap(unmatched);
+}
+
+void TriggerObjectStandAlone::unpackNamesAndLabels(const edm::EventBase & event,const edm::TriggerResults &res)
+{
+ unpackPathNames(event.triggerNames(res));
+ unpackFilterLabels(event,res);
+
+}
+
+void TriggerObjectStandAlone::unpackFilterLabels(const edm::EventBase & event,const edm::TriggerResults &res)  {
+  unpackFilterLabels(*allLabels(psetId_,event,res));
+}
+
+void TriggerObjectStandAlone::unpackFilterLabels(const std::vector<std::string> &labels)  {
+    if (filterLabelIndices_.empty()) return;
+
+    std::vector<std::string> mylabels(filterLabels_);
+    for (unsigned int i = 0, n = filterLabelIndices_.size(), m = labels.size(); i < n; ++i) {
+        if (filterLabelIndices_[i] >= m) throw cms::Exception("RuntimeError", "Error, filter label index out of bounds");
+        mylabels.push_back(labels[i]);
+    }
+    filterLabelIndices_.clear();
+    filterLabels_.swap(mylabels);
+}
+void TriggerObjectStandAlone::packFilterLabels(const edm::EventBase &event,const edm::TriggerResults &res)  {
+   packFilterLabels(*allLabels(psetId_,event,res));
+}
+
+
+void TriggerObjectStandAlone::packP4() {
+    setP4(reco::Particle::PolarLorentzVector(
+                MiniFloatConverter::reduceMantissaToNbitsRounding<14>(pt()),
+                MiniFloatConverter::reduceMantissaToNbitsRounding<11>(eta()),
+                MiniFloatConverter::reduceMantissaToNbits<11>(phi()),
+                MiniFloatConverter::reduceMantissaToNbitsRounding<8>(mass()) ));
```

Comment:
the reason for doing reduceMantissa here instead of the packing of PackedCandidate is that reduceMantissa is transparent both to the dataformat and to the reader: while packed candidates are always packed, trigger objects may not be so.